### DISCUSSION TITLE
RINGRI SQ-05 formatter board

### DISCUSSION THREAD

**Discussion by jekader** [mounaiban/captdriver#34] [Ideas]
I noticed that there are "alternative" formatter boards for sale in China for various GDI printers, including the LBP2900 ( RINGRI SQ-05), LJ1020 (RINGRI SQ-02), M1005 (RINGRI SQ-03) and others:

https://www.aliexpress.com/item/1005004827744403.html
![image](https://github.com/mounaiban/captdriver/assets/2088123/24467fbd-2373-41e3-a61b-1eba82799aa2)

https://www.aliexpress.com/item/1005003166309349.html
![image](https://github.com/mounaiban/captdriver/assets/2088123/4f54da73-359c-4e4d-a2f6-5debb3f1141c)

![image](https://github.com/mounaiban/captdriver/assets/2088123/3e1244a3-7308-4580-bf5c-55d8eb63925c)

Looks like these boards have nothing to do with the originals and are based on the STM32 MCU.
Did anyone have experience using these with captdriver? If they work and pretend to be the originals, this means the original firmware was reverse-engineered and ported to the STM32, which on its own allows for customization given the multitude of tools available. 

Since the LJ1020 for example has a Chinese clone (Zonewin ZY-1020), having a hackable formatter board opens up quite some potential in creating a laser printer with FOSS firmware.

**Comment by mounaiban** [comment]
:exploding_head: 

Thanks for bringing this to my attention!

I hope this thing is for real, as I would definitely get one if I can be sure it works like the original, and with the original Canon drivers. Captdriver is a near-subset of the original driver, if a printer with this third-party board works with the original drivers, then it can work with captdriver.

I'm amazed that someone even bothered to make a replacement board for what I thought to be an outdated (yet far from obsolete) device. I think I have underestimated the actual popularity of the LBP2900 and 3000.

I'd speculate that no newer printer (laser or inkjet) in the personal printing segment has yet matched the build quality, and definitely cost of ownership, of the small LBPs, to the extent that there is still a market for replacement parts for a late-2000s low-price printer in 2024.

Also, I have no evidence that the original boards _didn't_ use STM32s; the big square IC on the original board has non-standard markings, but the pinout looks suspiciously similar...
